targets/TARGET_ublox/TARGET_HI2110/objects.h

```diff
@@ -47,21 +47,26 @@ struct gpio_irq_s {
     uint32_t ch;   /* Corresponds to the interrupt pin */
 };
 
-struct serial_s {
-    SerialConfig config;
-    PinName rx_pin;
-    PinName tx_pin;
-    volatile uart_ctrl_t *reg_base;
-    uint8_t index;
+struct serial_s {    
+    PinName rx_pin; /* PinName will be used to find if UART1 or UART2 is addressed, PinNames, PortNames, PeripheralNames, Pinmap is yet to be updated */
+    PinName tx_pin; 
+    SerialConfig config; /* This enum is used to initialize the UART being addressed */	
+    volatile struct uart_s *reg_base;
+    uint8_t index; /* IRQ index number */
     uint32_t baud_rate;
+    uint16_t baud_rate_divider; /* calculated from baudrate */
+    uint8_t nco_phase; /* calculated from baudrate */
+    uint8_t flow_control; 
+    bool baudrate_gen_enable;
     bool format_set; /* If true then the struct that follows is populated */
```

Comment:
You might find that you don't need this struct.  I only needed it in HI2100 because, if the user wanted to switch to a baud rate greater than 9600 I had to entirely switch to another physical UART HW block and that required me to remember all the user settings so that I could reapply them.  Anyway, entirely up to you, but don't put stuff in here that you don't really need.

Reply:
I understand. I will generate another request with required changes. Thanks